libavcodec/jpeg2000dec.c

```diff
@@ -2155,6 +2319,31 @@ static int jpeg2000_read_main_headers(Jpeg2000DecoderContext *s)
             Jpeg2000Tile *tile;
             Jpeg2000TilePart *tp;
 
+
+            if (codsty->cblk_style & JPEG2000_CTSY_HTJ2K_F ){
+                Jpeg2000HTJ2KCodeStream stream;
+                /* Confirm Marker constraints according to Annex A of Rec. ITU-T T.814 | ISO/IEC 15444-15 */
+                // A.2
+                if (!(s->avctx->profile & (1<<14))){
+                    av_log(s->avctx, AV_LOG_ERROR, "Bit 14 of Rsiz is not equal to 1\n");
+                    return AVERROR_INVALIDDATA;
+                }
+                // A.3.1
+                if(!(s->pcap & (1 << (17)))){ // 15th MSB of Pcap
+                    av_log(s->avctx, AV_LOG_ERROR, "Bit 15 of Pcap is not equal to 1\n");
+                    return AVERROR_INVALIDDATA;
+                }
+                // A.5
+                for (int i=0;i<4;i++){
+                    if (s->roi_shift[i]>37){
+                        av_log(s->avctx, AV_LOG_ERROR, "ROI coefficient %d is greater than 37.\n",s->roi_shift[i]);
+                        return AVERROR_INVALIDDATA;
+                    }
+                }
+                if (jpeg2000_set_htj2k_constrains(s,&stream)){
```

Comment:
What about printing as `AV_LOG_INFO` the constrain sets to which the codestream belongs (for debugging)?